src/ProductView/TabBar/QueryInterface/QueryInterface.js

```diff
@@ -0,0 +1,143 @@
+import React, { Component } from 'react';
+import Modal from './Modal/Modal.js';
+import QueryInterfaceHeader from './QueryInterfaceHeader/QueryInterfaceHeader.js';
+
+class QueryInterface extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            eventTypeName: '',
+            eventTypeAttributes: [
+                {
+                    "id": 0,
+                    "name": "productId",
+                    "type": "INTEGER",
+                    "readonly": true
+                },
+                {
+                    "id": 1,
+                    "name": "productFamilyId",
+                    "type": "STRING",
+                    "readonly": true
+                },
+                {
+                    "id": 2,
+                    "name": "",
+                    "type": "",
+                    "readonly": false
+                }
+            ],
+            eventQuery: '',
+            validationResult: ''
+        };
+        this.nextAttributeId = 3;
+        this.handleChangeEventTypeName = this.handleChangeEventTypeName.bind(this);
+        this.handleChangeAttributeName = this.handleChangeAttributeName.bind(this);
+        this.handleChangeAttributeType = this.handleChangeAttributeType.bind(this);
+        this.handleChangeQuery = this.handleChangeQuery.bind(this);
+        this.handleSaveQuery = this.handleSaveQuery.bind(this);
+    }
+
+    componentDidMount() {
+        this.setState({ eventQuery: this.props.defaultQuery });
```

Comment:
Please do not use props for defining defaultQueries.
Transfer it to a static method and after merging it to developer transfer it to config.